### DISCUSSION TITLE
Mass beatmap import from Windows to Android

### DISCUSSION THREAD

**Discussion by kamild1996** [ppy/osu#24066] [Q&A]
I have 400-something maps in my osu!lazer installation on Windows, which I would like to import to osu!lazer on my phone, and I'm having trouble finding a good way to do it.
My first thing to try would be to copy the data folder of osu from desktop to phone, but Android has been blocking access to /Android/data for a while, so I don't think I can copy anything over this way.
How else could I possibly try to import all these maps without searching for them on osu!direct one by one?

**Comment by nagi-desuuu** [comment]
You can just copy the entire Lazer folder to your phone and it'll work the same, assuming that:
- Your phone isn't running on Android 13, which prevent direct access to `Android/data`
- Your phone's storage capacity is big enough to handle your Lazer folder
- You have the patience to transfer over what I can assume GBs of data from your computer to your phone

**Reply by kamild1996** [reply]
Yep, unfortunately my phone is running Android 13. No luck for me then?

**Reply by ItsZer0here** [reply]
there might still be a chance:

1. install shizuku
2. install FV file explorer
3. start shizuku with usb debugging OR wireless debugging (since you have an android version above 10)

done, now FV file explorer has access to Android/data

**Comment by peppy** [comment]
There's no easy way to do this other than what has been mentioned above.

**Comment by kamild1996** [comment]
Thank you for your responses @toadsworth-jr and @peppy.

I have found a way to do this using another external tool - [kabiiQ's BeatmapExporter](https://github.com/kabiiQ/BeatmapExporter). I used this tool to export beatmaps into .osz files, then copied the exported songs onto the phone and used a file manager app (eg. MiXplorer, but I imagine many file explorers will work) to select all these files and **share** them (not "open with") to osu. After a moment, the game started importing them.

In my case (Moto Edge 30 [Snapdragon 778G+] with Android 13), the importing process of 463 beatmaps took about 15 minutes to complete, and Android wasn't happy when it was happening for some reason, spamming me with "System interface not responding" messages. But it worked!

![Screenshot_20230628-181807](https://github.com/ppy/osu/assets/5320414/825bba2f-20c6-4291-bc58-9a819baf97f8)
![Screenshot_20230628-181929](https://github.com/ppy/osu/assets/5320414/23477b38-8acd-4618-a527-0c09dfc18aa7)

Hope this makeshift solution helps anyone, and that we will eventually get a better, official way to do this :)
